index.ts

```diff
@@ -168,14 +203,33 @@ class ServerlessCustomDomain {
      */
     public initializeVariables(): void {
         this.enabled = this.evaluateEnabled();
-        if (this.enabled) {
-            const credentials = this.serverless.providers.aws.getCredentials();
+        this.enabledWs = this.evaluateEnabledWs();
+
+        let credentials;
+
+        if (this.enabled || this.enabledWs) {
```

Comment:
Nitpicking here, but not a fan of the naming convention `something` and `somethingWs` - maybe specify that `something` refers to rest api endpoints as a direct contrast with `somethingWs`?